From 07e679b98c61c1c55a41629c7b4dfa1b8137a6ad Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 5 Sep 2023 23:01:23 -0500
Subject: [PATCH] Disable setlocalversion from running

If the source tree is a git tree (IE, when git bisecting) then setlocalversion runs and adds a `+` to the KERNELRELEASE variable.
This breaks booting, so don't do that and instead set the KERNELRELEASE to the KERNELVERSION
---
 Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index d0efd84bb7d0..4a444e3a7797 100644
--- a/Makefile
+++ b/Makefile
@@ -1253,11 +1253,7 @@ vmlinux: vmlinux.o $(KBUILD_LDS) modpost
 # make sure no implicit rule kicks in
 $(sort $(KBUILD_LDS) $(KBUILD_VMLINUX_OBJS) $(KBUILD_VMLINUX_LIBS)): . ;
 
-ifeq ($(origin KERNELRELEASE),file)
-filechk_kernel.release = $(srctree)/scripts/setlocalversion $(srctree)
-else
-filechk_kernel.release = echo $(KERNELRELEASE)
-endif
+filechk_kernel.release = echo $(KERNELVERSION)
 
 # Store (new) KERNELRELEASE string in include/config/kernel.release
 include/config/kernel.release: FORCE
-- 
2.42.0

